File: two_sum/two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i, item in enumerate(nums):
            if (target - item) not in nums:
                continue
            else:

                for j, item_j in enumerate(nums):
                    if i == j:
                        continue
                    elif item_j != (target - item):
                        continue
                    else:
                        return [i, j]
solution = Solution()
print(solution.twoSum([3, 2, 4], 6))

File: two_sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  var map = new Map();
  for (let i = 0; i < nums.length; i++) {
    if (map.get((target - nums[i])) != null) {
      return [i, map.get(target - nums[i])]
    }
    else {
      map.set(nums[i], i)
    }
  }
};
console.log(twoSum([3, 4, 9, 6, 4], 8 ))